src/grasp_filter.cpp

```diff
@@ -429,6 +429,47 @@ std::size_t GraspFilter::filterGraspsHelper(std::vector<GraspCandidatePtr>& gras
   return remaining_grasps;
 }
 
+void GraspFilter::preFilterBySuctionVoxelOverlap(std::vector<moveit_grasps::GraspCandidatePtr>& grasp_candidates,
+                                                 double cutoff)
+{
+  // We pre-filter by removing all candidates with less than some overlap with the desired object.
+  std::size_t grasp_candidates_before = grasp_candidates.size();
+  std::size_t count = 0;
+  for (std::size_t ix = 0; ix < grasp_candidates.size();)
+  {
+    ROS_DEBUG_STREAM_NAMED("grasp_filter.pre_filter", "-------------------------------------------------------");
+    ROS_DEBUG_STREAM_NAMED("grasp_filter.pre_filter", "Grasp candidate " << count++);
+    bool valid = false;
+    for (double& voxel_overlap : grasp_candidates[ix]->suction_voxel_overlap_)
+    {
+      ROS_DEBUG_STREAM_NAMED("grasp_filter.pre_filter", "cutoff: " << cutoff << "\tvoxel score " << voxel_overlap);
+      if (voxel_overlap > cutoff)
+      {
+        valid = true;
+        continue;
```

Comment:
I think this continue is redundant here, there is no code below this that could be skipped. A break would make sense though.

Reply:
good catch.